pkg/proxmox/goproxmox/api_client.go

```diff
@@ -39,29 +39,55 @@ var ErrVMIDFree = errors.New("VMID is free")
 // APIClient Proxmox API client object.
 type APIClient struct {
 	*proxmox.Client
-	logger logr.Logger
+	logger             logr.Logger
+	reserveOnlyRunning bool
+}
+
+type Option func(*APIClient)
+
+func WithCalcOnlyRunning() Option {
+	return func(c *APIClient) {
+		c.reserveOnlyRunning = true
+	}
 }
 
 // NewAPIClient initializes a Proxmox API client. If the client is misconfigured, an error is returned.
-func NewAPIClient(ctx context.Context, logger logr.Logger, baseURL string, options ...proxmox.Option) (*APIClient, error) {
+// options can be goproxmox.Option or proxmox.Option
+func NewAPIClient(ctx context.Context, logger logr.Logger, baseURL string, options ...any) (*APIClient, error) {
+	var pOpts []proxmox.Option
+	var apiOpts []Option
+	for _, option := range options {
+		switch o := option.(type) {
+		case proxmox.Option:
+			pOpts = append(pOpts, o)
+		case Option:
+			apiOpts = append(apiOpts, o)
+		default:
+			return nil, fmt.Errorf("invalid option %T", option)
+		}
+	}
```

Comment:
This should've been a type.

Reply:
Can you get example
How to not use any, if we have type from upstream library "proxmox.Option", and we need local type in this library?